rllib/algorithms/algorithm.py

```diff
@@ -2827,6 +3080,10 @@ def _before_evaluate(self):
         """Pre-evaluation callback."""
         pass
 
+    @DeveloperAPI
```

Comment:
nit: Let's remove this. We should replace these with custom callbacks (for users). If an algo needs to do something specific before evaluation, it can do so by overriding the `evaluate` method and then do this before calling `super.evaluate()` ...

Reply:
The corresponding `_before_evaluate` is currently NOT used by any algorithm.